test/externalTests/prb-math.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import sys
+
+from exttest.common import CURRENT_EVM_VERSION, InvalidConfigError
+from exttest.common import parse_command_line, run_test
+from exttest.common import TestConfig, SolcConfig
+
+from runners.foundry import FoundryRunner
+
+if __name__ == "__main__":
+    try:
+        args = parse_command_line("PRBMath external tests", sys.argv[1:])
+
+        runner_config = TestConfig(
+            repo_url="https://github.com/PaulRBerg/prb-math.git",
+            ref_type="branch",
+            ref="main",
+            build_dependency="rust",
+            compile_only_presets=[
+                "ir-no-optimize",
```

Comment:
Looks like there's some issue, either with this or with Foundry, because this preset is failing here while it works in `prb-maths.sh`. Or is it just that it compiled in `v2.5.0` but no longer compiles on `main`?

Reply:
Also, I see in test output that this is running with `london` EVM even though the default is now `shanghai`.

```
-------------------------------------
Settings preset: ir-no-optimize
Settings: {'optimizer': {'enabled': 'false', 'details': {'yul': 'false'}}, 'evmVersion': 'london', 'viaIR': 'true'}
EVM version: london
Compiler version: 0.8.21
Compiler version (full): 0.8.21-ci.2023.6.15+commit.77985a9a.Linux.g++
-------------------------------------
```

Reply:
> Looks like there's some issue, either with this or with Foundry, because [this preset is failing in this PR](https://app.circleci.com/pipelines/github/ethereum/solidity/30199/workflows/9f88ba31-9202-49ae-9d4e-480319db432a/jobs/1340944) while it works in `prb-maths.sh`. Or is it just that it compiled in `v2.5.0` but no longer compiles on `main`?

Yes, the `ir-no-optimize` and `ir-optimize-evm-only` are failing in the `main` branch of prb-math with stack-too-deep error. So I comment them for now and will investigate if there is anything we can do about it.

Reply:
> Also, I see in test output that this is running with `london` EVM even though the default is now `shanghai`.
> 
> ```
> -------------------------------------
> Settings preset: ir-no-optimize
> Settings: {'optimizer': {'enabled': 'false', 'details': {'yul': 'false'}}, 'evmVersion': 'london', 'viaIR': 'true'}
> EVM version: london
> Compiler version: 0.8.21
> Compiler version (full): 0.8.21-ci.2023.6.15+commit.77985a9a.Linux.g++
> -------------------------------------
> ```

Yes, it was wrong. I fixed it now.